common/flatpak-run.c

```diff
@@ -3092,8 +3092,94 @@ flatpak_run_add_app_info_args (GPtrArray      *argv_array,
   return TRUE;
 }
 
+static gboolean
+symlink_target_is_deployed (GFile *runtime_files, const char *symlink, const char *target)
+{
+  const char *runtime_files_path = NULL;
+  g_autofree char *deployed_target = NULL;
+
+  /* Manually check whether the symlink would point to a resolvable path inside the sandbox. */
+  runtime_files_path = flatpak_file_get_path_cached (runtime_files);
+  if (target[0] == '/')
+    {
+      /* Symlink points to an absolute path */
+      guint base_index = (g_strcmp0 (target, "/usr") == 0) ? 4 : 1;
+      deployed_target = g_strdup_printf ("%s/%s", runtime_files_path, &target[base_index]);
```

Comment:
So, if the symlink is /foo/bar, we then look in $runtime/foo/bar, which will be in /usr/foo/bar in the sandbox. That doesn't seem right.